File: wintest/ui/web/frontend/src/docs/press_key.ts
import type { StepDoc } from './types';

export const pressKeyDoc: StepDoc = {
  name: 'press_key',
  title: 'Press Key',
  summary: 'Press a single keyboard key.',
  description:
    'Presses and releases a single keyboard key. Use this for keys like Enter, Tab, Escape, arrow keys, function keys, etc. For key combinations (e.g. Ctrl+C), use the hotkey step instead.',
  parameters: [
    {
      name: 'key',
      type: 'string',
      required: true,
      description:
        'The key to press. Examples: "enter", "tab", "escape", "backspace", "delete", "up", "down", "left", "right", "f1" through "f12", "home", "end", "pageup", "pagedown".',
    },
  ],
  example:
    '- type: press_key\n  key: "enter"\n  description: "Press Enter to confirm"',
};

File: wintest/steps/press_key.py
"""Press-key step -- press a single keyboard key."""

from ._base import StepDefinition, FieldDef
from ..tasks.schema import StepResult


def validate(step, step_num):
    if not step.key:
        return [f"Step {step_num}: 'press_key' requires a 'key' field"]
    return []


def execute(step, agent):
    agent.actions.press_key(step.key)
    return StepResult(step=step, passed=True)


definition = StepDefinition(
    name="press_key",
    description="Press a single keyboard key (enter, tab, escape, ...)",
    fields=[FieldDef("key", "string", required=True)],
    validate=validate,
    execute=execute,
)
